File: 프로그래머스/0/120837. 개미 군단/개미 군단.js
function solution(hp) {
    var answer = 0;
    

    var ant1 = 5;
    var ant2 = 3;
    var ant3 = 1;

    var hp_1 = hp % ant1;

    var ant_num = parseInt(hp / ant1);

    if(hp_1 >= ant2) {
        ant_num += parseInt(hp_1 / ant2);
        hp_1 = hp_1 % ant2;

        if (hp_1 >= ant3) {
            ant_num += parseInt(hp_1 / ant3);
        }
    } else if (hp_1 >= ant3) {
        ant_num += parseInt(hp_1 / ant3);
    }

    answer = ant_num;
    
    return answer;
}

File: 프로그래머스/0/120837. 개미 군단/개미 군단.java
class Solution {
    public int solution(int hp) {
        int answer = 0;
        
        int ant1 = 5;
        int ant2 = 3;
        int ant3 = 1;
        
        int hp_1 = hp % ant1;
        
        int ant_num = hp / ant1;
        
        if(hp_1 >= ant2) {
            ant_num += hp_1 / ant2;
            hp_1 = hp_1 % ant2;
            
            if (hp_1 >= ant3) {
                ant_num += hp_1 / ant3;
            }
        } else if (hp_1 >= ant3) {
            ant_num += hp_1 / ant3;
        }
        
        answer = ant_num;
        
        return answer;
    }
}